metricbeat/scripts/docs_collector.py

```diff
@@ -19,20 +21,34 @@ def collect(beat_name):
 
     modules_list = {}
 
-    # Iterate over all modules
-    for module in sorted(os.listdir(base_dir)):
-
-        module_doc = path + "/" + module + "/_meta/docs.asciidoc"
+    modules_path = [{
+        'base_dir': oss_base_dir,
+        'path': oss_path + "/" + module,
+        'name': module,
+        'metricsets': sorted(os.listdir(oss_path + "/" + module))
+    } for module in filter(lambda module: os.path.isfile(oss_path + "/" + module + "/_meta/docs.asciidoc"),
+                           os.listdir(oss_base_dir))]
+
+    if os.path.isdir(os.path.abspath(xpack_base_dir)):
+        modules_path += [{
+            'base_dir': xpack_base_dir,
+            'path': xpack_path + "/" + module,
+            'name': module,
+            'metricsets': sorted(os.listdir(xpack_path + "/" + module)),
+        } for module in filter(lambda module: os.path.isfile(xpack_path + "/" + module + "/_meta/docs.asciidoc"),
+                               os.listdir(xpack_path))]
```

Comment:
The code to generate `modules_path` is twice here and only depends on `X_base_dir` (and `X_path` but can be obtained from the other one). Extract it to a function?